### PR TITLE
Add a 2nd argument to create()

### PR DESCRIPTION
**Pull request by ismail1432**
Maybe I'm wrong but I tried the exemple with my case and the test failed because IMO if we don't pass an object to second argument, $form->getData() will return an array and this->assertEquals() can't compare an object with an array.
What do you think :-)

### CODE REVIEW

**Comment by javiereguiluz** [comment]
Let's ask for a review to our Forms expert @vudaltsov Thanks!

**Comment by xabbuh** [comment]
To be more precise, the type of the returned data depends on whether or not the `data_class` option is set (see http://symfony.com/doc/current/reference/forms/types/form.html#empty-data). However, passing the data object when creating the form does not harm. So let's do this change. 👍

File: form/unit_testing.rst [hunk 1]
```diff
@@ -46,17 +46,21 @@ The simplest ``TypeTestCase`` implementation looks like the following::
                 'test' => 'test',
                 'test2' => 'test2',
             );
-
-            $form = $this->factory->create(TestedType::class);
-
+           
+           $objectToCompare = new TestObject();
+           //$objectToCompare will retrieve data from the form submission pass it at second argument
```
Comment (xabbuh) [inline form/unit_testing.rst]: missing space after `//`

File: form/unit_testing.rst [hunk 2]
```diff
@@ -46,17 +46,21 @@ The simplest ``TypeTestCase`` implementation looks like the following::
...
+            $form = $this->factory->create(TestedType::class, $objectToCompare);
+            
             $object = new TestObject();
             // ...populate $object properties with the data stored in $formData
 
             // submit the data to the form directly
             $form->submit($formData);
-
+            
+            //Will fill data in $objectToCompare
```
Comment (xabbuh) [inline form/unit_testing.rst]: same here

File: form/unit_testing.rst [hunk 3]
```diff
@@ -46,17 +46,21 @@ The simplest ``TypeTestCase`` implementation looks like the following::
...
+            $objectToCompare = $form->getData();
```
Comment (xabbuh) [inline form/unit_testing.rst]: I would not introduce this variable here. There is no need for it.

File: form/unit_testing.rst (hunk 2 above)
Comment (xabbuh) [reply form/unit_testing.rst]: I would actually just remove this comment

File: form/unit_testing.rst (hunk 1 above)
Comment (vudaltsov) [reply form/unit_testing.rst]: and 1 space before `//`, 1 space before `$objectToCompare` (the indents are incorrect).

File: form/unit_testing.rst [hunk 4]
```diff
@@ -46,17 +46,21 @@ The simplest ``TypeTestCase`` implementation looks like the following::
...
+           //$objectToCompare will retrieve data from the form submission pass it at second argument
+            $form = $this->factory->create(TestedType::class, $objectToCompare);
+            
             $object = new TestObject();
             // ...populate $object properties with the data stored in $formData
 
             // submit the data to the form directly
             $form->submit($formData);
-
+
```
Comment (vudaltsov) [inline form/unit_testing.rst]: remove spaces on an empty line

**Review by vudaltsov — CHANGES_REQUESTED** [review]
I like this PR, it's a common practice to use object itself, not `$form->getData()`.

But CS should be fixed.

**Comment by ismail1432** [comment]
Thanks for all feedbacks, I'll work on quickly,

File: form/unit_testing.rst (hunk 3 above)
Comment (ismail1432) [reply form/unit_testing.rst]: I understand but IMO it's strange to declare a variable, pass it as an argument and never use it after.

**Comment by wouterj** [comment]
:+1: Looks great, thanks!

status: reviewed

**Comment by javiereguiluz** [comment]
Thanks for this nice contribution! Help with form docs is always much appreciated! We merged this on 2.7 branch.

**Comment by ismail1432** [comment]
It’s always a pleasure to help us even I’m still waiting for the « Documentation’s Badge contributor » 😜
Enjoy Guys !

File: form/unit_testing.rst (hunk 3 above)
Comment (xabbuh) [reply form/unit_testing.rst]: But now we are doing exactly that because we are overwriting the previously declared variable.

Comment (javiereguiluz) [reply form/unit_testing.rst]: @xabbuh do we need to change anything in the merged PR? If yes, could you please create a quick issue (or PR) to not forget about it? Thanks a lot!

Comment (xabbuh) [reply form/unit_testing.rst]: see #9431

**Pull request closed without merging**